File: core/src/main/java/pl/alpheratzteam/obfuscator/transformer/DebugInfoRemoverTransformer.java
package pl.alpheratzteam.obfuscator.transformer;

import org.objectweb.asm.ClassReader;
import org.objectweb.asm.ClassWriter;
import org.objectweb.asm.tree.ClassNode;
import pl.alpheratzteam.obfuscator.Obfuscator;
import java.util.HashMap;
import java.util.Map;

public class DebugInfoRemoverTransformer extends Transformer
{
    public DebugInfoRemoverTransformer(Obfuscator obfuscator) {
        super(obfuscator);
    }

    @Override
    public void visit(Map<String, ClassNode> classMap) {
        final Map<String, ClassNode> map = new HashMap<>();
        classMap.values().forEach(classNode -> {
            final ClassWriter writer = new ClassWriter(ClassWriter.COMPUTE_MAXS);
            classNode.accept(writer);

            final ClassNode newClassNode = new ClassNode();
            new ClassReader(writer.toByteArray()).accept(newClassNode, ClassReader.SKIP_DEBUG);
            map.put(newClassNode.name, newClassNode);
        });

        classMap.clear();
        classMap.putAll(map);
    }
}

File: src/main/kotlin/pl/alpheratzteam/obfuscator/transformer/DebugInfoRemoverTransformer.kt
package pl.alpheratzteam.obfuscator.transformer

import org.objectweb.asm.ClassReader
import org.objectweb.asm.ClassWriter
import org.objectweb.asm.tree.ClassNode
import pl.alpheratzteam.obfuscator.Obfuscator
import pl.alpheratzteam.obfuscator.api.transformer.Transformer

/**
 * @author Unix
 * @since 18.12.2020
 */

class DebugInfoRemoverTransformer : Transformer {

    override fun transform(obfuscator: Obfuscator) {
        val classes = mutableMapOf<String, ClassNode>()
        obfuscator.classes.values.forEach {
            val classWriter = ClassWriter(ClassWriter.COMPUTE_MAXS)
            it.accept(classWriter)

            val newClassNode = ClassNode()
            ClassReader(classWriter.toByteArray()).accept(newClassNode, ClassReader.SKIP_DEBUG)
            classes[newClassNode.name] = newClassNode
        }

        obfuscator.classes.clear()
        obfuscator.classes.putAll(classes)
    }

}